### DISCUSSION TITLE
action calling PressureAction

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#22567] [Q&A General] [answered]
(Re-routing this discussion from Slack since @permcody is busy)

I'm developing a MOOSE App that has an action that calls PressureAction from TensorMechanics (see my code [here](https://github.com/jessecarterMOOSE/my_pressure_demo/blob/678dd4356a93b49805978ad7bbcb3b4f9327a48b/src/actions/MyPressureAction.C#L34-L63)). When I run an input file containing a block for the action, the action runs from what I can tell (cout statements), but doesn't actually apply the BC that I'm ultimately trying to do. I'm new to writing actions, so it's probably something simple. I'm just copying what I see elsewhere in the framework.

**Comment by GiudGiud** [comment, marked as answer]
Hello

The "action" runs, do you mean the meta_action adding the Pressure one or do you mean the PressureAction?

Do you have an input with this?

Guillaume

**Reply by jessecarterMOOSE** [reply]
My action runs - I can tell because I have a bunch of print statements that I see in the output. Can't say for sure if the PressureAction is being run. My app has an input file [here](https://github.com/jessecarterMOOSE/my_pressure_demo/blob/678dd4356a93b49805978ad7bbcb3b4f9327a48b/test/tests/actions/my_pressure_action/my_pressure_action.i). There's another in the same directory where I use PressureAction instead of my action to get the expected outcome.

**Reply by GiudGiud** [reply]
ok looked into the code but have not run it yet
Could you please try this:

You did not set the displacement parameter. In the pressure action, the pressure BC are only added if there are displacements
```
  // Create pressure BCs
  for (unsigned int i = 0; i < displacements.size(); ++i)
  {
```

**Reply by GiudGiud** [reply]
I ll make a PR to make displacements required. @dschwen agreed

**Reply by jessecarterMOOSE** [reply]
But I have those in GlobalParams

**Reply by GiudGiud** [reply]
Yeah I dont know how this pans out. I can try to run it

**Reply by GiudGiud** [reply]
I just added these to the parameters in your action and got the expected behavior.

Let s keep track of this, it seems global params did not get applied to the action, even though set-global_params ran after your custom action

**Reply by jessecarterMOOSE** [reply]
Oh, I see, I need to have my action take in "displacements" parameter and pass it on to PressureAction. That does it!

**Comment by jessecarterMOOSE** [comment]
[Here's the updated code](https://github.com/jessecarterMOOSE/my_pressure_demo/blob/89d6fe95ef090fb1c23eaa23d8558c62cf6def5b/src/actions/MyPressureAction.C)
